PyInstaller/compat.py

```diff
@@ -178,69 +176,23 @@
         from win32ctypes.pywin32 import pywintypes  # noqa: F401
         from win32ctypes.pywin32 import win32api
     except ImportError:
-        # This environment variable is set by seutp.py
+        # This environment variable is set by setup.py
         # - It's not an error for pywin32 to not be installed at that point
         if not os.environ.get('PYINSTALLER_NO_PYWIN32_FAILURE'):
             raise SystemExit('PyInstaller cannot check for assembly dependencies.\n'
                              'Please install pywin32-ctypes.\n\n'
                              'pip install pywin32-ctypes\n')
 
 
-def _architecture():
-    """
-    Returns the bit depth of the python interpreter's architecture as
-    a string ('32bit' or '64bit'). Similar to platform.architecture(),
-    but with fixes for universal binaries on MacOS.
-    """
-    if is_darwin:
-        # Darwin's platform.architecture() is buggy and always
-        # returns "64bit" event for the 32bit version of Python's
-        # universal binary. So we roll out our own (that works
-        # on Darwin).
-        if sys.maxsize > 2 ** 32:
-            return '64bit'
-        else:
-            return '32bit'
-    else:
-        return platform.architecture()[0]
-
-architecture = _architecture()
-del _architecture
-
-def _system():
-    # On some Windows installation (Python 2.4) platform.system() is
-    # broken and incorrectly returns 'Microsoft' instead of 'Windows'.
-    # http://mail.python.org/pipermail/patches/2007-June/022947.html
-    syst = platform.system()
-    if syst == 'Microsoft':
-        return 'Windows'
-    return syst
-
-system = _system()
-del _system
-
-def _machine():
-    """
-    Return machine suffix to use in directory name when looking
-    for bootloader.
-
-    PyInstaller is reported to work even on ARM architecture. For that
-    case `system` and `architecture` are not enough.
-    Path to bootloader has to be composed from `system`, `architecture`
-    and `machine` like:
-        'Linux-32bit-arm'
-    """
-    mach = platform.machine()
-    if mach.startswith('arm'):
-        return 'arm'
-    elif mach.startswith('aarch'):
-        return 'aarch'
-    else:
-        # Assume x86/x86_64 machine.
-        return None
+# macOS's platform.architecture() can be buggy, so we do this manually there.
```

Comment:
Two points:

1. I'd like to preserve the comments here -- it's useful explanation
2. Is this still true? Or does current Python fix this?

Reply:
https://docs.python.org/3/library/platform.html

Yes. Well kind of.

I'll add some more comments.